### PR TITLE
v3 agent config docs update

### PR DESCRIPTION
**Pull request by nijikokun**
Resolves ngrok-private/ngrok#33235

Ready for review.

Tasks:

- [x] Endpoint Options
- [x] Link Configuration Field Sections
- [x] Write supplementary details for the Examples
- [x] Tunnels (Deprecation)
- [x] Example for randomly assigned addresses
- [x] Migration

Future:

- [ ] Full Configuration File (similar to v2)

### CODE REVIEW

**Comment by KristopherPaulsen** [comment]
Giving this a looky-lu! Mostly just double checking / giving my unrequested opinion on wordsmithing (which you should feel free to ignore 🤣).

File: docs/agent/config/v3.mdx [hunk 1]
```diff
@@ -462,11 +249,387 @@ These are a list of specifiers for what `Host` headers will be allowed to make r
...
+    upstream:
+      url: 8080
+      protocol: http1
+      proxy_protocol: 2
+# ...
+```
+
+## Example Configuration Files
+
+Below are a collection of different agent configurations to serve as examples for your `ngrok.yml` file. The [next section](/docs/agent/config/v3/#endpoint-configurations) explains how to start some or all of your configured endpoints from the CLI.
```
Comment (KristopherPaulsen) [inline docs/agent/config/v3.mdx:544]: I think this was originally my copy-text / _my fault_, this link doesn't seem to go anywhere : |

**Review by KristopherPaulsen — COMMENTED** [review]
✅ Links look good (one small dead link)
✅ Examples look good 🎉 

I think we got "most"™️ of the fiddly bits covered when we chatted, but ping me on slack when you want a final lookover. These changes are 👨‍🍳 💋

File: docs/agent/config/v3.mdx [hunk 2]
```diff
@@ -462,11 +249,387 @@ These are a list of specifiers for what `Host` headers will be allowed to make r
...
+```sh
+ngrok start --all
+```
+
+### Endpoint Configuration Options
+
+| Name                                                 | Required | Protocols   | Description                                                                                                                                                            |
+| ---------------------------------------------------- | -------- | ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`name`](#name)                                      | Required |             | A unique name for this endpoint's configuration.                                                                                                                       |
+| [`url`](#url)                                        | Required |             | The address you would like to use to forward traffic to your upstream service.                                                                                         |
```
Comment (KristopherPaulsen) [inline docs/agent/config/v3.mdx:308]: Okay, I might _also_ have mislead you here as well. You _can_ omit the `url` for the endpoint (and you'll receive a random one)

eg

```
version: 3 
agent:
  authtoken: PSYCH!
endpoints:
  - name: heyThisCanBeEmptyButIt'llBeAHTTPSEndpoint
    upstream:
      url: 2323

```

**Review by justngrok — COMMENTED** [review]
looks good, lots of changes, can review again, but in the interest of getting this updated ASAP, approving.

**Review by justngrok — APPROVED** [review]

**Pull request merged**